### ISSUE TITLE
CPLEX binaries should not be mandatory for Docker build

### ISSUE THREAD

**Issue by psalvy** [EPFL-LCSB/pytfa#5]
See #2

**Comment by psalvy** [comment]
@Midnighter in #2 you referred to the Docker build unable to complete because of missing CPLEX files? I was not able to reproduced this behavior. However, It did require a gurobi.lic file, provided as gurobi.lic.template. Was it the issue you experienced ? If so commit [69c479e](https://github.com/EPFL-LCSB/pytfa/commit/69c479eccc0ceb939a539bf678e722be4ecc0f8a) should address this
Cheers,
Pierre

**Comment by Midnighter** [comment]
Let me try again. Should I rather work with master or the dev branch?

**Comment by psalvy** [comment]
Thanks :) master is fine, the PR #6 should have fixed the issue.

**Comment by Midnighter** [comment]
So I don't set up any solver and just try to run `docker build .` which leads to a permission error. My impression was that the default solver would be GLPK only using other solvers if they are available. Maybe that is wrong and either CPLEX or GUROBI are required. My `docker/solvers/` directory only contains the instructions.

```
Sending build context to Docker daemon  29.18kB                                               
Step 1/27 : FROM python:3.5
 ---> 1aa6b224c0e1                                                                            
Step 2/27 : USER root                                                                         
 ---> Using cache                                                                             
 ---> 36f7a915280b                                                                            
Step 3/27 : RUN apt-get update && apt-get install -y --no-install-recommends         libxml2-d
ev             libxslt1-dev             less                        && rm -rf /var/lib/apt/lis
ts/*                                                                                          
 ---> Using cache                                                                             
 ---> daf5d755a47c                                                                            
Step 4/27 : ENV USER pytfa                                                                    
 ---> Using cache                                                                             
 ---> 087c39a87062                                                                            
Step 5/27 : ENV HOME /home/$USER                                                              
 ---> Using cache      
 ---> a06987c590b9                                                                            
Step 6/27 : RUN useradd -ms "/bin/bash" "$USER"                                               
 ---> Using cache                                                                             
 ---> d6abd8ef49ae                                                                            
Step 7/27 : USER $USER                                                                        
 ---> Using cache                                                                             
 ---> bd58ea182c79
Step 8/27 : WORKDIR $HOME
 ---> Using cache
 ---> 01fed92cb27f
Step 9/27 : USER root
 ---> Using cache
 ---> ab0227bf2c92
Step 10/27 : COPY requirements.txt .
 ---> Using cache
 ---> 4e9cdfc14127
Step 11/27 : RUN pip install -r requirements.txt
 ---> Using cache
 ---> 6d3c2464aa9b
Step 12/27 : COPY ./solvers /solvers
 ---> Using cache
 ---> 43684ab5d845
Step 13/27 : COPY ./utils /utils
 ---> Using cache
 ---> 91fec73f9093
Step 14/27 : RUN /utils/install_cplex.sh
 ---> Running in 32be21c39e28
/bin/sh: 1: /utils/install_cplex.sh: Permission denied
The command '/bin/sh -c /utils/install_cplex.sh' returned a non-zero code: 126
```

**Comment by psalvy** [comment]
Oh, I see. The expected behavior is indeed that CPLEX and Gurobi are optional, and GLPK is the default solver, you are not mistaken on that :)

The solver install scripts are executed but include an if condition looking for the solver. If the files are not found, the script exits without doing anything. 

The issue is that git does not transfer permissions, and if you look at the permissions of the .sh files you'll see execution is not allowed. I added to the Dockerfile a line that explicitly gives execution permissions. Can you try the dev branch and let me know if that fixes your issue ?

**Comment by psalvy** [comment]
Closed in #8
